### PR TITLE
also run macos arm64 in merge group

### PR DESCRIPTION
**Pull request by mgovers**
Relates to #479 .

We cannot enable MacOS arm64 testing on PRs, because PRs have multiple (even up til quite a lot of) reruns/commits/... with a successful CI run. That would eventually result in our total amount of free minutes being used up for FlyCI.

Since merge queue CI runs typically only run once for each PR, we may "abuse" it to do this additional check before final release to `main` without prohibitively increasing the risk of going over the limit for free minutes on FlyCI runners.

### CODE REVIEW

**Review by nitbharambe — APPROVED** [review]
Nice idea

**Comment by Jerry-Jinfeng-Guo** [comment]
What is the difference of doing it like this comparing to the previous way? Does it replace the previous way?

**Comment by mgovers** [comment]
> What is the difference of doing it like this comparing to the previous way? Does it replace the previous way?

it's an additional check. before, the test would only happen on `main` after the rest of CI and before releasing. `main` would be broken if the test would fail.

Now, it's checked on the merge queue in addition to `main`, which means that we can catch issues earlier in the process

**Comment by Jerry-Jinfeng-Guo** [comment]
> > What is the difference of doing it like this comparing to the previous way? Does it replace the previous way?
> 
> it's an additional check. before, the test would only happen on `main` after the rest of CI and before releasing. `main` would be broken if the test would fail.
> 
> Now, it's checked on the merge queue in addition to `main`, which means that we can catch issues earlier in the process

I am very much not sure if that is a good idea though... it we are far from consuming the free runs, sure; but otherwise, we are most of the time doing the same check twice. Perhaps wait until Tony's input to merge?

**Comment by mgovers** [comment]
> > > What is the difference of doing it like this comparing to the previous way? Does it replace the previous way?
> > 
> > 
> > it's an additional check. before, the test would only happen on `main` after the rest of CI and before releasing. `main` would be broken if the test would fail.
> > Now, it's checked on the merge queue in addition to `main`, which means that we can catch issues earlier in the process
> 
> I am very much not sure if that is a good idea though... if we are far from consuming the free runs, sure; but otherwise, we are most of the time doing the same check twice. Perhaps wait until Tony's input to merge?

we indeed are far from consuming the free runs. @TonyXiang8787 did an investigation before we introduced this CI job in which he concluded that we are barely in the danger zone, even if we would enable it for PRs. A single run of the job costs about 45s, while we have >500min available per month for free (taken from https://www.flyci.net/pricing#features ).

That means that we would have to have like 600 CI runs, or 300 PR merges per month, before we go over our limits. I feel like we're safe for the time being

**Pull request merged**
